mmengine/fileio/io.py

```diff
@@ -102,48 +904,33 @@ def dump(obj, file=None, file_format=None, file_client_args=None, **kwargs):
     if file_format not in file_handlers:
         raise TypeError(f'Unsupported format: {file_format}')
 
+    if file_client_args is not None:
+        warnings.warn(
+            '"file_client_args" will be deprecated in future. '
+            'Please use "backend_args" instead', DeprecationWarning)
+        if backend_args is not None:
+            raise ValueError(
+                '"file_client_args and "backend_args" cannot be both set.')
+
     handler = file_handlers[file_format]
     if file is None:
         return handler.dump_to_str(obj, **kwargs)
     elif is_str(file):
-        file_client = FileClient.infer_client(file_client_args, file)
+        if file_client_args is not None:
+            file_client = FileClient.infer_client(file_client_args, file)
```

Comment:
does file_client_args have the same key values as those in backend_args?
If they have, we may consider using an existing function named deprecation_warning to directly map the file_client_args to backend_args.

Reply:
If not, we should put backend_args before file_client_args since it is an argument to be deprecated.

Reply:
> does file_client_args have the same key values as those in backend_args? If they have, we may consider using an existing function named deprecation_warning to directly map the file_client_args to backend_args.

file_client_args has an extra key 'prefix'

Reply:
> If not, we should put backend_args before file_client_args since it is an argument to be deprecated.

Changing the order of parameters is to cause bc-breaking.